### PR TITLE
[NBS Docs] Update integrate search into plugins docs

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

Closes: https://github.com/backstage/backstage/issues/26239

Update the "Integrate search into plugins" tutorial to use the New Backend System.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/plugins/integrating-search-into-plugins.md [hunk 1]
```diff
@@ -24,27 +28,53 @@ The search platform provides an interface (`DocumentCollatorFactory` from packag
 
 > You can always look at a working example, e.g. [StackOverflowQuestionsCollatorFactory](https://github.com/backstage/backstage/blob/master/plugins/search-backend-module-stack-overflow-collator/src/collators/StackOverflowQuestionsCollatorFactory.ts), if you are unsure or want to follow best practices.
 
-#### 1. Install collator interface dependencies
+#### 1. Create a collator module package
 
-We will need the interface `DocumentCollatorFactory` from package `@backstage/plugin-search-common`, so let's add it to your plugins dependencies:
+In order to add a FAQ collator to the Backstage index registry, we have to create a [plugin module](https://backstage.io/docs/backend-system/building-plugins-and-modules/index#modules) and the best way to do this is to create it in a separate package, e.g., `plugins/search-backend-module-faq-snippets-collator`, using the `backstage-cli new` command:
```
Comment (awanlin) [inline docs/plugins/integrating-search-into-plugins.md:33]:
```suggestion
In order to add a FAQ collator to the Backstage index registry, we have to create a [plugin module](https://backstage.io/docs/backend-system/building-plugins-and-modules/index#modules) and the best way to do this is to create it in a separate package, e.g., `plugins/search-backend-module-faq-snippets-collator`, using the `yarn new` command:
```

File: docs/plugins/integrating-search-into-plugins.md [hunk 2]
```diff
@@ -24,27 +28,53 @@ The search platform provides an interface (`DocumentCollatorFactory` from packag
...
 
-```sh
-# navigate to the plugin directory
-# (for this tutorial our plugin lives in the backstage repo, if your plugin lives in a separate repo you need to clone that first)
-cd plugins/faq-snippets
+1. Access your Backstage project root directory and run `npx @backstage/cli new`;
```
Comment (awanlin) [inline docs/plugins/integrating-search-into-plugins.md:35]:
```suggestion
1. Access your Backstage project root directory and run `yarn new`;
```

File: docs/plugins/integrating-search-into-plugins.md [hunk 3]
```diff
@@ -24,27 +28,53 @@ The search platform provides an interface (`DocumentCollatorFactory` from packag
...
+
+We will use some libraries in the module, so let's add them to your plugin module dependencies:
 
+```sh
 # Create a new branch using Git command-line
 git checkout -b tutorials/new-faq-snippets-collator
 
 # Install the package containing the interface
-yarn add @backstage/plugin-search-common
+yarn workspace backstage-plugin-search-backend-module-faq-snippets-collator add cross-fetch @backstage/config @backstage/plugin-search-common @backstage/plugin-search-backend-node
```
Comment (awanlin) [inline docs/plugins/integrating-search-into-plugins.md:49]:
```suggestion
yarn workspace backstage-plugin-search-backend-module-faq-snippets-collator add node-fetch @backstage/config @backstage/plugin-search-common @backstage/plugin-search-backend-node
```

Just thinking of what we decided in the ADR: https://backstage.io/docs/architecture-decisions/adrs-adr013

File: docs/plugins/integrating-search-into-plugins.md [hunk 4]
```diff
@@ -80,52 +101,48 @@ Imagine your FAQs can be retrieved at the URL `https://backstage.example.biz/faq
 }
 ```
 
-Below we provide an example implementation of how the FAQ collator factory could look like using our new document type, placed in the `plugins/faq-snippets/src/search/collators/FaqCollatorFactory.ts` file:
+Below we provide an example implementation of how the FAQ collator factory could look like using our new document type, placed in the `plugins/search-backend-module-faq-snippets-collator/src/factory.ts` file:
 
 ```ts
 import fetch from 'cross-fetch';
```
Comment (awanlin) [inline docs/plugins/integrating-search-into-plugins.md:107]:
```suggestion
import fetch from 'node-fetch';
```

File: docs/plugins/integrating-search-into-plugins.md [hunk 5]
```diff
@@ -138,13 +155,107 @@ export class FaqCollatorFactory implements DocumentCollatorFactory {
...
+
+After that, install the module on your Backstage backend instance:
+
+```ts title='packages/backend/src/index.ts'
+import { createBackend } from '@backstage/backend-defaults';
+//...
+const backend = createBackend();
+// Installing the search backend plugin
+backend.add(import('@backstage/plugin-search-backend/alpha'));
+// Installing the newrly created faq snippets collator module
```
Comment (awanlin) [inline docs/plugins/integrating-search-into-plugins.md:227]:
```suggestion
// Installing the newly created faq snippets collator module
```

**Review by awanlin — CHANGES_REQUESTED** [review]
🎉 awesome work @camilaibs, thanks for taking this on! Over looks great just needs a few minor adjustments.

**Comment by camilaibs** [comment]
Thanks for the suggestions @awanlin, could you please take a look again 🙏🏻 ?

File: docs/plugins/integrating-search-into-plugins.md [hunk 6]
```diff
@@ -138,13 +155,109 @@ export class FaqCollatorFactory implements DocumentCollatorFactory {
...
+      },
+    });
+  },
+});
+```
+
+In the fragment above, the module is registered, and when the Backstage backend initializes it, it adds the FAQ Snippets collator to the search index registry. Now let's export the module as default from the `index.ts` file:
+
+```ts title='plugins/search-backend-module-faq-snippets-collator/src/index.ts'
+export { export const searchFaqSnippetsCollatorModule as default } from './module';
```
Comment (freben) [inline docs/plugins/integrating-search-into-plugins.md:208]:
```suggestion
export { searchFaqSnippetsCollatorModule as default } from './module';
```

File: docs/plugins/integrating-search-into-plugins.md [hunk 7]
```diff
@@ -138,13 +155,109 @@ export class FaqCollatorFactory implements DocumentCollatorFactory {
...
+```
+
+#### 7. Install the collator module
+
+The newly created module should be added to the backend package dependencies as follows:
+
+```sh
+yarn --cwd backend add @internal/backstage-plugin-search-backend-module-faq-snippets-collator
```
Comment (freben) [inline docs/plugins/integrating-search-into-plugins.md:216]:
```suggestion
yarn --cwd packages/backend add @internal/backstage-plugin-search-backend-module-faq-snippets-collator
```

right?

File: docs/plugins/integrating-search-into-plugins.md [hunk 8]
```diff
@@ -138,13 +155,109 @@ export class FaqCollatorFactory implements DocumentCollatorFactory {
...
+```
+
+#### 8. Testing the collator code
 
 To verify your implementation works as expected make sure to add tests for it. For your convenience, there is the [`TestPipeline`](https://backstage.io/docs/reference/plugin-search-backend-node.testpipeline) utility that emulates a pipeline into which you can integrate your custom collator.
 
 Look at [DefaultTechDocsCollatorFactory test](https://github.com/backstage/backstage/blob/de294ce5c410c9eb56da6870a1fab795268f60e3/plugins/techdocs-backend/src/search/DefaultTechDocsCollatorFactory.test.ts), for an example.
 
-#### 6. Make your plugins collator discoverable for others
+You can also check out the documentation on [how to test Backstage plugin modules](https://backstage.io/docs/backend-system/building-plugins-and-modules/testing).
```
Comment (freben) [inline docs/plugins/integrating-search-into-plugins.md:243]: should this be a relative path instead of a link, like the others?

Comment (camilaibs) [reply docs/plugins/integrating-search-into-plugins.md]: Good catch, fixed 🙏🏻

File: docs/plugins/integrating-search-into-plugins.md [hunk 9]
```diff
@@ -80,52 +101,48 @@ Imagine your FAQs can be retrieved at the URL `https://backstage.example.biz/faq
...
     this.logger = options.logger;
   }
 
-  static fromConfig(config: Config, options: FaqCollatorFactoryOptions) {
-    const baseUrl =
-      config.getOptionalString('faq.baseUrl') ||
-      'https://backstage.example.biz/faq-snippets';
-    return new FaqCollatorFactory({ ...options, baseUrl });
+  static fromConfig(
+    config: Config,
```
[the remark this answers is not in the document]
Comment (camilaibs) [reply docs/plugins/integrating-search-into-plugins.md]: Fixed

**Review by freben — APPROVED** [review]

**Review by awanlin — APPROVED** [review]

**Pull request merged**